pkg/nodediscovery/nodediscovery.go

```diff
@@ -176,7 +176,7 @@ func (n *NodeDiscovery) StartDiscovery(nodeName string) {
 	n.LocalNode.EncryptionKey = node.GetIPsecKeyIdentity()
 	n.LocalNode.WireguardPubKey = node.GetWireguardPubKey()
 	n.LocalNode.Labels = node.GetLabels()
-	n.LocalNode.NodeIdentity = uint32(identity.ReservedIdentityHost)
+	n.LocalNode.NodeIdentity = identity.GetLocalNodeID().Uint32()
```

Comment:
I have a suspicion that this will break kube-apiserver policy in the case where the kube-apiserver resides on the same node as the pod, logically I guess that's when a pod has such a `toEntities: kube-apiserver` policy and attempts to contact the kube-apiserver (on the same host) using that policy.  As far as I can tell, we don't have coverage of this path in `test/k8sT/Policies.go`.

Reply:
^^ I'm OK with the revert for v1.9/v1.10 though. Just saying that for v1.11/later it's worth looking a bit closer.

Reply:
Context: https://github.com/cilium/cilium/pull/17823#discussion_r746832701

(Original discussion was at the end of Paul's review, screenshot included below for your convenience)
![image](https://user-images.githubusercontent.com/1243336/153514068-7c07c5f3-4863-488a-811a-9e90c90c87c4.png)